lib/pages/search/view/search_page.dart

```diff
@@ -0,0 +1,299 @@
+import 'package:acs_upb_mobile/pages/classes/service/class_provider.dart';
+import 'package:acs_upb_mobile/pages/classes/view/class_view.dart';
+import 'package:acs_upb_mobile/pages/people/model/person.dart';
+import 'package:acs_upb_mobile/pages/people/service/person_provider.dart';
+import 'package:acs_upb_mobile/pages/people/view/person_view.dart';
+import 'package:acs_upb_mobile/pages/search/view/seached_classes_view.dart';
+import 'package:acs_upb_mobile/widgets/scaffold.dart';
+import 'package:acs_upb_mobile/widgets/search_bar.dart';
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class SearchPage extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _SearchPageState();
+}
+
+class _SearchPageState extends State<SearchPage> {
+  String query = '';
+  List<ClassHeader> classesSearched;
+  bool searchClosed = true;
+  List<Person> peopleSearched;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AppScaffold(
+      title: Text(S.current.navigationSearch),
+      body: ListView(
+        scrollDirection: Axis.vertical,
+        children: [
+          SearchWidget(
+            onSearch: (searchText) => {setState(() => query = searchText)},
+            cancelCallback: () => {setState(() => query = '')},
+            searchClosed: false,
+          ),
+          if (query.isNotEmpty)
+            FutureBuilder(
+                future: Provider.of<PersonProvider>(context, listen: false)
+                    .search(query),
+                builder: (_, snapshot) {
+                  if (snapshot.connectionState == ConnectionState.done) {
+                    peopleSearched = snapshot.data;
+                    if (peopleSearched.isNotEmpty) {
+                      return Column(
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          Padding(
+                            padding: const EdgeInsets.fromLTRB(10, 10, 0, 10),
+                            child: Text(
+                              S.current.labelPeople,
+                              style: const TextStyle(fontSize: 15),
+                            ),
+                          ),
+                          PeopleCircleList(
+                            people: peopleSearched,
+                            query: query,
+                          )
+                        ],
+                      );
+                    }
+                    return Container();
+                  } else {
+                    return Container();
+                  }
+                }),
+          const Divider(
+            color: Colors.white,
+          ),
+          if (query.isNotEmpty)
+            FutureBuilder(
```

Comment:
This method got way too long, make the code more modular. Create a separate widget for each section (`ClassesSearchResults(query)`, `PeopleSearchResults(query)`). You can change the `PeopleCircleList` and `ClassesCircleList` widgets below - take the build method there, put the content in another helper method (e.g. `buildList()`) and make it part of the bigger widget that contains the entire search result.